src/bundle/Security/TwoFactor/Provider/TwoFactorProviderPreparationListener.php

```diff
@@ -106,7 +106,11 @@ public function onTwoFactorForm(TwoFactorAuthenticationEvent $event): void
 
     public function onKernelResponse(ResponseEvent $event): void
     {
-        if (!$event->isMasterRequest()) {
+        if (method_exists(KernelEvents::class, 'isMainRequest') && !$event->isMainRequest()) {
+            return;
+        }
+
+        if (!$event->isMainRequest()) {
```

Comment:
Now you're using only `isMainRequest`, but you need to fall back to `isMasterRequest` when `isMainRequest` isn't available.